src/dataServices/dynamoDbDataService.ts

```diff
@@ -94,13 +98,26 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
 
     async createResource(request: CreateResourceRequest) {
         const { resourceType, resource, id } = request;
+
         const vid = 1;
         let item = resource;
         item.resourceType = resourceType;
         item.meta = generateMeta(vid.toString());
 
         const params = DynamoDbParamBuilder.buildPutAvailableItemParam(item, id || uuidv4(), vid);
-        await this.dynamoDb.putItem(params).promise();
+        try {
+            await this.dynamoDb.putItem(params).promise();
+        } catch (e) {
+            if (e.code === 'ConditionalCheckFailedException') {
+                // It is highly unlikely that the autogenerated id will collide with a preexisting id, but adding that case
+                // to cover all bases
+                const errorMessage = id
+                    ? `Unable to create resource with id ${id}, because resource already exists`
+                    : 'Auto generated id matched an existing id, please try again';
+                throw new InvalidResourceError(errorMessage);

```

Comment:
Related PR to handle this error
https://github.com/awslabs/fhir-works-on-aws-routing/pull/52